src/core_plugins/interpreter/init.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { routes } from './server/routes';
+import { functionsRegistry } from '@kbn/interpreter-utils/common/lib/functions_registry';
+import { loadServerPlugins } from './server/lib/load_server_plugins';
+
+export default function (server /*options*/) {
+  server.injectUiAppVars('canvas', () => {
+    const config = server.config();
+    const basePath = config.get('server.basePath');
+    const reportingBrowserType = config.get('xpack.reporting.capture.browser.type');
+
+    return {
+      kbnIndex: config.get('kibana.index'),
+      esShardTimeout: config.get('elasticsearch.shardTimeout'),
+      esApiVersion: config.get('elasticsearch.apiVersion'),
+      serverFunctions: functionsRegistry.toArray(),
+      basePath,
+      reportingBrowserType,
+    };
+  });
+
+  loadServerPlugins();
```

Comment:
all server plugins can be loaded synchronously here, no need to wait as we need to in x-pack/canvas